kernel/core/datamanagement/memory.cpp

```diff
@@ -3926,8 +3928,7 @@ KMeansResult Memory::getKMeans(const VectorIndex& columns, size_t nClusters, siz
     std::vector<std::vector<mu::value_type>> centroids;
 
     // Random generator
-    mt19937 mt(time(nullptr));
-
+    mt19937 mt = getRandGenInstance();

```

Comment:
```suggestion
    std::mt19937& mt = getRandGenInstance();
```
Reference it, do not copy.

Reply:
So it is an unnecessary copy operation, right? Or are there more differences ?
I did not look into the RandomGenerator, surely a CopyConstructor could use a new Seed and I would get different random numbers.